0015-helm-value-style-variable-interfaces/README.md

```diff
@@ -0,0 +1,215 @@
+
+### Goals
+
+- Simplify the use of Zarf Variables for members of the community familiar with Helm
+
+### Non-Goals
+
+- Entirely redesign Zarf Variables, Constants and Templates
+  - (while this is a non-goal for _this_ proposal it is worth discussion if this would better serve the goal above)
+
+## Proposal
+
+The proposed solution is to change the internal tracking of Zarf Variables from a `map[string]string` to a `map[string]interface{}` as well as allowing the `map[string]interface{}` to be loaded from a `zarf-config` file (including all of the existing formats that the file accepts).
+
+Additionally `--set` on the CLI would align to the [Helm `--set` syntax](https://helm.sh/docs/intro/using_helm/#the-format-and-limitations-of---set) with the values on the right-hand side of the `=` being handled as described there instead of as a `string`.
+
+### User Stories (Optional)
+
+#### Story 1
+
+**As** Jacquline **I want** to be able to set full Helm value objects in a Zarf config variable **so that** I can simplify package configurations and rely on my existing familiarity with Helm.
+
+**Given** I have a Zarf Package with a Helm value override in a chart
+**When** I deploy that package with a `zarf-config.yaml` like the below*:
+```yaml
+package:
+  deploy:
+    set:
+      MY_OVERRIDE_VARIABLE_NAME:
+        key: value
+        bool: true
+        num: 100
+        arr: []
+        obj: {}
+```
+**Then** Zarf will apply the entire interface to the Helm chart override
+
+> [!NOTE]
+> *This would apply to all `zarf-config` formats not just YAML
+
+#### Story 2
+
+**As** Jacquline **I want** to be able to set full Helm value objects from the CLI **so that** I can simplify package configurations and rely on my existing familiarity with Helm.
+
+**Given** I have a Zarf Package with a Helm value override in a chart
+**When** I deploy that package with a `--set` like the below:
+```yaml
+zarf package deploy zarf-package-test.tar.zst --set MY_OVERRIDE_VARIABLE_NAME.key=value,MY_OVERRIDE_VARIABLE_NAME.bool=true,MY_OVERRIDE_VARIABLE_NAME.num=100,MY_OVERRIDE_VARIABLE_NAME.arr=[]
+```
+**Then** Zarf will apply the entire interface to the Helm chart override
+
+> [!NOTE]
+> *This should follow the [Helm `--set` syntax](https://helm.sh/docs/intro/using_helm/#the-format-and-limitations-of---set)
+
+### Risks and Mitigations
+
+This will require some additional processing of the `zarf-config` files to allow them to be processed properly because of Viper's long-running [case-insensitive keys issue](https://github.com/spf13/viper/issues/1014).  This could be performed similar to UDS CLI's implementation of this feature and / or by deprecating some of the existing less-used formats for configuration. (TODO: (@AustinAbro321) - link this to the other ZEP for viper reconsideration)
```

Comment:
FYI we're not going to create a ZEP to evaluate viper at this time since there are higher priorities, but we are tentatively planning to deprecate and remove all the config file options besides yaml and toml.

Reply:
updated